server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -385,6 +385,31 @@ public Consumer update(Consumer updatedConsumer) {
     }
 
     /**
+     * @param updatedConsumer updated Consumer values.
+     * @return Updated consumers
+     */
+    @Transactional
+    public Consumer updateNoFlush(Consumer updatedConsumer) {
+        Consumer existingConsumer = find(updatedConsumer.getId());
+        if (existingConsumer == null) {
+            return create(updatedConsumer);
+        }
+
+        // TODO: Are any of these read-only?
```

Comment:
@mstead @vritant this was just a copy paste from regular  'update' method. In master, there is the same TODO comment. I am actually not sure what the author of the TODO meant, so I kept the TODO in the code.